### PR TITLE
Fit refactor

### PR DESCRIPTION
**Pull request by rafa-guedes**
Refactor method fit_jonswap and fit_gaussian so they can return both (or either) of the fit spectra or the parameters.

### CODE REVIEW

**Review by pbranson — CHANGES_REQUESTED** [review]
@rafa-guedes I think this is a much better interface for the fitting. Thanks for these updates.

A couple of minor comments for you to consider.

File: wavespectra/specarray.py [hunk 1]
```diff
@@ -967,104 +962,96 @@ def rmse(self, other):
         rmse = ediff / e0sum
         return rmse
 
-    def fit_jonswap(self, smooth=True, gamma0=1.5):
+    def fit_jonswap(self, smooth=True, gamma0=1.5, spectra=True, params=True):
```
[the remark this answers is not in the document]
Comment (rafa-guedes) [reply wavespectra/specarray.py]: Makes sense, this is removed now

File: wavespectra/specarray.py [hunk 2]
```diff
@@ -967,104 +962,96 @@ def rmse(self, other):
...
+            dsout["hs"] = hs
+            dsout["gamma"] = gamma
         set_spec_attributes(dsout)
-        dsout.attrs.update(self._get_cf_attributes(attrs.SPEC1DNAME))
-        return dsout.rename(attrs.SPEC1DNAME)
+        dsout.attrs["title"] = "Fit Jonswap spectra"
+        return dsout
 
-    def fit_gw(self, smooth=True, gw0=1.5):
+    def fit_gaussian(self, smooth=True, gw0=1.5, spectra=True, params=True):
```
[the remark this answers is not in the document]
Comment (rafa-guedes) [reply wavespectra/specarray.py]: Done

File: wavespectra/core/fitting.py [hunk 3]
```diff
@@ -133,6 +134,7 @@ def fit_gaussian_spectra(ef, freq, fp0, hs0, gw0):
     return npstats.gaussian(freq, fp, hs, gw)
 
 
-def fit_gaussian_width(ef, freq, fp0, hs0, gw0):
+def fit_gaussian_params(ef, freq, fp0, hs0, gw0):
     """Wrapper to return only gw from _fit_gaussian to run as ufunc."""
```
[the remark this answers is not in the document]
Comment (rafa-guedes) [reply wavespectra/core/fitting.py]: Done

**Comment by rafa-guedes** [comment]
Thanks Paul, I have made the changes you suggested and will merge this. We are missing tests (and docs) for the fitting capability but we can add those later on.

**Pull request merged**